examples/gethostlatency.rs

```diff
@@ -0,0 +1,128 @@
+use bcc::perf_event::init_perf_map;
+use bcc::BccError;
+use bcc::{Uprobe, Uretprobe, BPF};
+use clap::{App, Arg};
+
+use core::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use std::{fmt, ptr};
+
+/*
+ * Basic Rust clone of `gethostlatency`, from the iovisor tools.
+ * https://github.com/iovisor/bcc/blob/master/tools/gethostlatency.py
+ *
+ * Prints the latency for getaddrinfo/gethostbyname(3) calls, which are a common part of DNS lookups.
+ */
+
+/*
+ * Define the struct the BPF code writes in Rust
+ * This must match the struct in `gethostlatency.c` exactly.
+ * The important thing to understand about the code in `gethostlatency.c` is that it creates structs of
+ * type `latency_event_t` and pushes them into a buffer where our Rust code can read them.
+ */
+#[repr(C)]
+struct latency_event_t {
+    pid: u32,
+    delta: u64,
+    comm: [u8; 16], // TASK_COMM_LEN
+    host: [u8; 80],
+}
+
+impl fmt::Display for latency_event_t {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "{:<9} {:<6} {:<16} {:>10.2} {}",
+            chrono::Utc::now().format("%T"),
+            self.pid,
+            get_string(&self.comm),
+            (self.delta as f32) / 1e6,
+            get_string(&self.host)
+        )
+    }
+}
+
+fn do_main(runnable: Arc<AtomicBool>) -> Result<(), BccError> {
+    let matches = App::new("gethostlatency")
+        .about("Prints the latency for getaddrinfo/gethostbyname(2) calls")
+        .arg(
+            Arg::with_name("pid")
+                .long("pid")
+                .short("p")
+                .help("Only trace calls from this PID")
+                .takes_value(true),
+        )
+        .get_matches();
+
+    let pid: Option<i32> = matches
+        .value_of("pid")
+        .map(|v| v.parse::<i32>().expect("Invalid argument for PID"));
+
+    // attach entry and return probes for common host lookup functions
+    let libc_so_path = "/lib/x86_64-linux-gnu/libc.so.6";
+    let code = include_str!("gethostlatency.c");
+    let mut module = BPF::new(code)?;
+    for symbol in vec!["getaddrinfo", "gethostbyname", "gethostbyname2"] {
+        Uprobe::new()
+            .handler("do_entry")
+            .binary(libc_so_path)
+            .symbol(symbol)
+            .pid(pid)
+            .attach(&mut module)?;
+        Uretprobe::new()
+            .handler("do_return")
+            .binary(libc_so_path)
+            .symbol(symbol)
+            .pid(pid)
+            .attach(&mut module)?;
+    }
+
+    // the "events" table is where the "function was called" events get sent
+    let table = module.table("events")?;
+    // install a callback to print out latency events when they happen
+    let mut perf_map = init_perf_map(table, perf_data_callback)?;
```

Comment:
`init_perf_map` [will be be deprecated in favour of `PerfMapBuilder::new(table, cb).build()`](https://github.com/rust-bpf/rust-bcc/blob/5053fc9a4863c0556a1959a52edb4a1da1dbb793/src/perf_event/perf_map.rs#L72)

Reply:
Agree. I would definitely prefer the new style be used in examples in this repo. We will mark `init_perf_map()` as deprecated at some point. I just opened #142 to track that.